### ISSUE TITLE
Toggling WebRTC setting doesn't work when same setting is enabled in uBlock

### ISSUE THREAD

**Issue by ablanathtanalba** [EFForg/privacybadger#2528]
Per a client support email. I am able to reproduce this issue in Firefox on MacOS. I first suspected this might be related to #2499, though with some manual testing I am still able to reproduce the issue without those code changes. I am not able to reproduce this buggy behavior on the uBlock side of things (when the PB setting is toggled, I can still change the same setting in the uBlock options). I'm happy to look into fixing this issue this week.

**Comment by ghostwords** [comment]
This is working as intended, no? We disable the WebRTC setting if we detect it's controlled by a different extension (or rather, [re-enable if we detect it's controlled by us](https://github.com/EFForg/privacybadger/blob/7290134618de15ce1cd66137b2c7ba73b860e1e3/src/js/options.js#L125-L127)).

**Comment by ablanathtanalba** [comment]
If it's working as intended, maybe I could put something in place that wouldn't mislead a user to think that the leaking their IP through webRTC wasn't already protected against? A few options that come to mind:

* have the radio selected by default if PB options page detects that webRTC leaking protection is enforced via some other mechanism (which could then be misleading if they try to deselect it...)
* remove the option from appearing on the PB options page if it's both protected against via some other mechanism (like uBlock) and not already selected by the user through PB
* disable the radio button and display a message saying they're already protected against it

**Comment by ablanathtanalba** [comment]
I [https://github.com/EFForg/privacybadger/tree/clear-webrtc-setting-msg](made a branch) that does something like the last option, something I just quickly slapped together. It would make the options page look like this if the webrtc ip leak is protected via something other than pb:

![Screen Shot 2019-12-19 at 9 46 08 AM](https://user-images.githubusercontent.com/25778052/71196898-47264680-2245-11ea-929f-6f057f4b14b1.png)

**Comment by ghostwords** [comment]
I like having the checkbox checked, if the value of the WebRTC setting is equal to or higher than whatever value we set it to. If you'd like to make that tweak, open a PR and I'll review.

Adding explanatory text that says something to the effect of "this setting is controlled by another extension" is also good, but it involves translations, and I'm just not sure it's worth the effort at this time (as we haven't gotten much feedback about this behavior yet).

In conclusion, I'm fine with doing nothing here for now, or with improving the logic that determines whether the checkbox should be checked or not.

**Comment by ghostwords** [comment]
I think I see what happened here ... uBlock Origin's WebRTC checkbox [sets the underlying setting to "disable_non_proxied_udp" in Chrome but to "default_public_interface_only" in Firefox](https://github.com/gorhill/uBlock/blob/9637ab56a938460fe18a2217217e4eb4bdca5b28/platform/chromium/vapi-background.js#L181-L183).

We always set it to "disable_non_proxied_udp". So there is no mismatch in Chrome, but there is in Firefox.

I misremembered, we do use the strictest setting. Apparently, [it may be too strict in Firefox](https://github.com/gorhill/uBlock/issues/3009#issuecomment-329798696), where the setting works differently from Chrome.

We should probably tweak our logic to match uBlock Origin's. I'm not aware of any (Firefox) user complaints at this time though, so let's hold off on doing any more work on this.

**Comment by gorhill** [comment]
@ghostwords 

I changed the code in uBO to use `disable_non_proxied_udp` since it's now fixed since Firefox 70.

**Comment by zdat** [comment]
I'm unable to turn this setting off. (Check box is reapplied after refreshing)

I'd rather have it controlled by uBlock, but I can't because Privacy Badger wont let me disable theirs.

I thought maybe it was reapplying because it was detecting that WebRTC is being protected (by uBlock), but even after turning it off in uBlock it still wont stay off.

Any help?

**Comment by ghostwords** [comment]
@zdat Privacy Badger currently checks to see if this setting is controlled by another extension (such as uBlock) and marks the checkbox as checked off and disabled (not modifiable) in that case. As this is clearly confusing, we are removing this extra logic in the next update.

As for why you still see the setting as enabled in Privacy Badger after you turn it off in uBlock, I'm guessing either the browser remembers that you had it enabled in Privacy Badger, or you have another extensions that also controls this setting. Does unchecking the checkbox in Privacy Badger and reloading its options page should the setting as disabled?

**Comment by zdat** [comment]
Oh ok, well I'm fine with it being checked then :)

I just remembered I have many about:config alterations in firefox and I am pretty sure WebRTC is protected there, I forgot about that. I disabled both extensions and it was still protected so yes it's probably there (or failing that a different extension but I'm 90% sure it's the about:config tweaks)

Thanks for the reply.
